### ISSUE TITLE
Wierd API behaviour cannot pin point why

### ISSUE THREAD

**Issue by MarcHagen** [MerginMaps/server#247]
I've quite some projects in my community version 2024.2.2.
No fancy or special configs.

The UI shows all my projects no problem.
But when navigating to a project some return with "Project not found"

Digging bit deeper the behavior get more interesting.
The call to `/v1/project/mergin/inmeten` seems to fail with 404
But the call to `/v1/project/by_uuid/0756627a-8984-45b7-8fe7-bcc748648a0f`, which is the same project, works and returns all data.

Could you help me pinpoint the problem here?

**Comment by MarcelGeo** [comment]
Thanks for feedback @MarcHagen  ... we are working on fix - It should be fixed in next release of CE

**Comment by MarcHagen** [comment]
Thank you @MarcelGeo !
Any plan when to release next CE?

**Comment by harminius** [comment]
@MarcHagen Can you give me more info about the issue to be able to replicate it?

Did you deleted, moved or renamed the project?
Was the project public?
Are the broken projects somehow different from the other projects?

The issue might be already fixed as we did some changes. CE will be released in the autumn.

**Comment by MarcHagen** [comment]
I've only updated mergin half year ago or so to the latest CE version.
All of the sudden I got a notification of the engineers that some projects are not loading properly 

> Did you deleted, moved or renamed the project?

No

> Was the project public?

Private projects within the CE scope and assigned to members.
But admins also cannot view them.

> Are the broken projects somehow different from the other projects?

All projects are mixed with gis files and images/photos taken on site

If you want more data or information, please let me know.
I can run commands if that helps, or provide dumps (not the whole db)

**Comment by MarcHagen** [comment]
@harminius 
It seems to be a database issue. 

The `project` table is filled with data.
But when I try to select a project by name, if fails

```sql
postgres.public> SELECT t.*
                 FROM public.project t
                 LIMIT 501
[2024-08-19 14:38:00] 33 rows retrieved starting from 1 in 206 ms (execution: 193 ms, fetching: 13 ms)
postgres.public> SELECT t.*
                 FROM public.project t
                 WHERE name='daan_Inmeten'
                 LIMIT 501
[2024-08-19 14:38:23] 0 rows retrieved in 37 ms (execution: 29 ms, fetching: 8 ms)
postgres.public> SELECT t.*
                 FROM public.project t
                 WHERE name like 'daan_Inmeten'
                 LIMIT 501
[2024-08-19 14:39:11] 1 row retrieved starting from 1 in 41 ms (execution: 33 ms, fetching: 8 ms)
```

**Comment by harminius** [comment]
@MarcHagen 
This is quite interesting!
I found `_` (underscore) has a special behaviour with `LIKE` operator. Good to know. It could be potentially reason of your results.
1, Can you double check `daan_Inmeten` string is exactly the same in both queries? (I see it looks the same but things are not always as they seem, you know)
2, Could you upload also the result of the queries?

**Comment by MarcHagen** [comment]
@harminius it gets even wilder.
So I've manually updated the `name` field to be `daan_Inmeten` and the web GUI, the app and qgis started to work again.
Except some hours later, the engineer reported it was not working anymore …

But I cannot replicate the situation anymore. 
Syncing and creating from qgis plugin and app seem to work  fine now for newer projects.

However, I'm not a postgres magician, so for the life of me I don't know how to prove this in any way.
And if I create a database dump to file, possibly the seeming weird `_` is transformed to UTF-8 or something.
The database dump all underscores are 0x5F _, but I don't know if the table name is different tho.

To be really honest, we renamed all our projects to not have an underscore.
Maybe something went wrong with the workspace removal in some CE versions back as then all projects got prepended with the `{workspace}_`.
